lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts

```diff
@@ -96,6 +101,15 @@ export abstract class BaseManagedIdentitySource {
             }
         }
 
+        // Azure Arc
+        if (response.status === HttpStatus.UNAUTHORIZED) {
```

Comment:
It seems that handling the response is specific to Azure ARC. Since you went down the road of inheritance, like Neha did in .NET, the solution would be to have "HandleResponse" as protected and for AzureArc to ADD (not replace) to the HandleResponse functionality.

The code in .NET is: 

```
HandleResponse(HttpResponse response)
{
     if (response.StatusCode == System.Net.HttpStatusCode.Unauthorized)
    { 
          newHttpResponse = HandleComplexRetryLogic();
          base.HandleResponse(newHttpResponse);
    }
    else {
      base.HandleResponse(newHttpResponse);
}

}
   
}


```

Reply:
I've implemented this feedback, but with a twist. It's a little different than how dotnet implements it, but I've begun to implement a "retry policy". and moved this logic inside of it. Please let me know your thoughts.